src/backend/server.js

```diff
@@ -131,6 +132,55 @@ app.post('/log', wrap(async (req, res) => {
   res.send('')
 }))
 
+app.post('/password_reset', wrap(async (req, res) => {
+  // All errors in this endpoint will result in a 200 response.
+  // This prevents people from mining emails through brute
+  // force.
+  if(!req.body.email) {
+    return res.status(200).end();
+  }
+
+  let user = await models.User.findOne({ where: {email:req.body.email} });
+
+  if(!user) {
+    return res.status(200).end();
+  }
+
+  console.log(req.body.email)
+  console.log(JSON.stringify(user))
+
+  // Generate a globally unique reset token for the user
+  // We create it as a property of the user since we only
+  // issue one at a time, and require that the token
+  // match the email address when updating.
+  await user.set('resetToken',uuid.v4());
```

Comment:
I'm not sure if `=` triggers any custom `set` functions on the model. It may. If it does, then there is no difference.

As for await, I was wanting to make sure that the event loop didn't get released inside the function when updating the model. I'm still fuzzy on how the es6 stuff works so this may not be necessary. I just wanted to avoid a race condition between `user.set` and `user.save` and, since they returned promises, I figured better safe then sorry...